scalr/data_source_environment.go

```diff
@@ -0,0 +1,116 @@
+package scalr
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	scalr "github.com/scalr/go-scalr"
+)
+
+func dataSourceScalrEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceEnvironmentRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cost_estimation_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"email": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"full_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloud_credentials": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"policy_groups": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		}}
+}
+
+func dataSourceEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	scalrClient := meta.(*scalr.Client)
+
+	envID := d.Get("id").(string)
+
+	log.Printf("[DEBUG] Read configuration of environment: %s", envID)
+
+	environment, err := scalrClient.Environments.Read(ctx, envID)
+	if err != nil {
```

Comment:
Need to return an error if it's not scalr.ErrResourceNotFound just like in workspaces (SCALRCORE-17490):
https://github.com/Scalr/terraform-provider-scalr/blob/develop/scalr/data_source_workspace.go#L105-L111

Reply:
fixed